src/pages/transactions/components/ListBox.tsx

```diff
@@ -0,0 +1,164 @@
+    minAmount: 0,
+    maxAmount: 0,
+    startDate: '',
+    endDate: '',
+  });
+
+  const [clientId, setClientId] = useState<string>('');
+
+  const { data: clientsResponse } = useClientsQuery();
+
+  const { data: resourceResponse } = useTransactionMatchResourceQuery(
+    props.dataKey,
+    searchParams.searchTerm,
+    clientId
+  );
+
+  const [resourceItems, setResourceItems] = useState<ResourceItem[]>();
+
+  const [clients, setClients] = useState<Client[]>();
+
+  const [isFilterModalOpened, setIsFilterModalOpened] =
+    useState<boolean>(false);
+
+  const [isInvoicesDataKey, setIsInvoicesDataKey] = useState<boolean>(false);
+
+  const isItemChecked = (itemId: string) => {
+    return Boolean(props.selectedIds?.find((id) => id === itemId)?.length);
+  };
+
+  const selectItem = (itemId: string, clientId?: string) => {
+    setClientId(clientId || '');
+
+    const filteredItemIdsList = props.selectedIds?.find((id) => itemId === id);
+
+    let updatedItemIds;
+
+    if (filteredItemIdsList?.length) {
+      updatedItemIds = props.selectedIds?.filter((id) => id !== itemId);
+      props.setSelectedIds(updatedItemIds);
+    } else {
+      updatedItemIds = [...(props.selectedIds || []), itemId];
+      props.setSelectedIds(updatedItemIds);
+    }
+  };
+
+  const getClientName = (clientId: string) => {
+    return clients?.find(({ id }) => id === clientId)?.name;
+  };
+
+  const getFormattedResourceList = (resourceList: any) => {
+    return resourceList?.map((resourceItem: any) => ({
+      id: resourceItem.id,
+      number: resourceItem.number,
+      name: resourceItem.name,
+      clientName: getClientName(resourceItem.client_id),
+      status_id: resourceItem.status_id,
+      amount: resourceItem.amount,
+      date: resourceItem.date,
+      client_id: resourceItem.client_id,
+    }));
+  };
+
+  useEffect(() => {
+    setClients(clientsResponse?.data.data);
+
+    setResourceItems(getFormattedResourceList(resourceResponse));
+
+    if (props.dataKey === 'invoices') {
+      setIsInvoicesDataKey(true);
+    }
+  }, [props.dataKey, resourceResponse, clientsResponse]);
+
+  useEffect(() => {
+    if (isInvoicesDataKey && !props.selectedIds?.length) {
+      setClientId('');
+    }
+  }, [props.selectedIds]);
+
+  return (
+    <div className="flex flex-col w-full">
+      <div
+        className={`flex justify-center px-5 py-3 relative border-b border-t border-gray-200 ${props.className}`}
+      >
+        <SearchArea
+          dataKey={props.dataKey}
+          isInvoicesDataKey={isInvoicesDataKey}
+          searchParams={searchParams}
+          setIsFilterModalOpened={setIsFilterModalOpened}
+          isFilterModalOpened={isFilterModalOpened}
+          setSearchParams={setSearchParams}
+          setSelectedIds={props.setSelectedIds}
+        />
+      </div>
+      <ul
+        className="flex flex-col justify-start overflow-y-auto"
+        style={{
+          height: isInvoicesDataKey ? 400 : 200,
+        }}
```

Comment:
Can we use Tailwind here?

Reply:
Actually, they use the same way of handling this lists with ul element. https://tailwindui.com/components/application-ui/lists/stacked-lists

In our case I must to limit the height because of the screen size and behaviour on the screen if we get overflow.